settings.local.example.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * @file
+ * Local Backdrop CMS configuration file.
+ */
+
+/**
+ * Database settings. Use only one of these and comment out or remove the other.
+ */
+
+// If you are not running CiviCRM, you can use the simple database setting.
+$database = 'mysql://username:password@127.0.0.1/backdropcmsorg';
+
+// If you are running CiviCRM, you must use the advanced database setting, which
+// includes prefixes that make the CiviCRM database tables visible to Backdrop.
+// See https://backdropcms.org/project/civicrm_table_prefixes for more
+// information about this.
+require_once __DIR__ . '/civicrm_table_prefixes.php';
+$databases = array(
+  'default' => array (
+    'default' =>
+    array (
+      'driver' => 'mysql',
+      'database' => 'backdropcmsorg',
+      'username' => '',
+      'password' => '',
+      'host' => '127.0.0.1',
+      'prefix' => $civicrm_table_prefixes,
```

Comment:
Below is my train of thought...  not comments on the documentation!

```
// If you are running CiviCRM, you must use the advanced database setting, which
// includes prefixes that make the CiviCRM database tables visible to Backdrop.
```

This doesn't make any sense. Changing the name of the tables in one database shouldn't do anything about Backdrop's ability to access data in a completely different database.

The CiviCRM database doesn't even (appear to) use prefixes on it's own tables. Unless the prefix is defined someplace else?
```
/**
 * Backdrop database connection.
 */
define( 'CIVICRM_UF_DSN', 'mysql://username:password@localhost/backdropcmsorg?new_link=true');
/**
 * CiviCRM database connection.
 */
define('CIVICRM_DSN', 'mysql://username:password@localhost/civicrm?new_link=true');
```

Doing more reading on https://backdropcms.org/project/civicrm_table_prefixes...

> creates prefixes in Backdrop for the CiviCRM tables

The civicrm tables are in the civicrm databse, right? That's what this module is for? Adding prefixes to the Backdrop database will not change the civi tables in a different database. 

...Unless, there are also civi tables in the backdrop database? That wouldn't make any sense. Why would there be civicrm tables in the Backdrop database if civi is supposed to be running in a separate database? What's the point of having a separate database if civi is still going to put its data in to the backdrop database?

> ... so that CiviCRM fields may be used in Backdrop db queries, enabling things like using CiviCRM fields in Backdrop Views.

Backdrop (and thus views) will automatically use the prefix, if provided, and will not, if not. Having one makes no difference. 

But I'm still baffled as to why there would be any civi data in the Backdrop database at all. Civi should be writing this data to its own database. Views should be able to query the civi database, as long as it knows the database exists, and knows how to join to it. 

For Backdrop to know there is a second database, our databases array **should** look something like what's below:
```
require_once __DIR__ . '/civicrm_table_prefixes.php';
$databases = array(
  'default' => array (
    'default' => array (
      'driver' => 'mysql',
      'database' => 'backdropdatabase',
      'username' => 'user123',
      'password' => 'pass123',
      'host' => '127.0.0.1',
      'prefix' => '',
      'charset' => 'utf8mb4',
      'collation' => 'utf8mb4_general_ci',
    ),
    'civicrm' => array (
      'driver' => 'mysql',
      'database' => 'civicrmdatabase',
      'username' => 'user123civi',
      'password' => 'pass123civi',
      'host' => '127.0.0.1',
      'prefix' => $civicrm_table_prefixes,
      'charset' => 'utf8mb4',
      'collation' => 'utf8mb4_general_ci',
    ),
  ),
);
```
And as long as CIVI provides a Views relationship between the Backdrop data and the civi data, it'll query it wherever it needs to be joined.  No prefixes are needed for either database -- but especially not civi prefixes on the backdrop database. 

This sounds like a fun rabbit hole for another day.  :)

Reply:
>  No prefixes are needed for either database -- but especially not civi prefixes on the backdrop database.

OK, try creating a View of users that contains a CiviCRM field (e.g., Display Name) without the prefixes and let me know how it goes. 😉

What each prefix is doing is telling Backdrop that when you make a `db_query()` like, for example, `SELECT * FROM {civicrm_contact}`, it will resolve (via the curly brackets) to this actual query: `SELECT * FROM civicrm.civicrm_contact`, i.e., go get that table from the `civicrm` database, rather than attempting to get it from the default `backdropcmsorg.civicrm_contact`, which, of course, does not exist.

It's useful if you want to use Views that display results that include fields from both Backdrop and CiviCRM.